### PR TITLE
added swarm monitor to keep track of swarm progress

### PR DESCRIPTION
**Pull request by wittjosiah**
follow-on will add termination criteria to monitor

### CODE REVIEW

File: VERSION [hunk 1]
```diff
@@ -1 +1 @@
-0.1.0
+0.1.1
```
Comment (kiambogo) [inline VERSION]: Do we need to bump the version? I feel like keep it at 0.1.0 until we're ready to actually use it

Comment (wittjosiah) [inline VERSION]: if we aren't going to change the version when we add things we might as well just call it 1.0-SNAPSHOT the whole time

Comment (kiambogo) [inline VERSION]: +1 to that

File: src/main/scala/SwarmMonitor.scala [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  solution: List[Double],
+  lbest: List[Double],
+  best: List[Double]
+) extends WorkerState
+case class AntState(
+  solution: List[Int],
+  best: List[Int]
+) extends WorkerState
+
+class SwarmMonitor(objectiveFunction: List[Any] => Double,
```
Comment (kiambogo) [inline src/main/scala/SwarmMonitor.scala]: SwarmMaster maybe? Fine either way, just master is more common in clusters

File: src/test/scala/SwarmMonitorSpec.scala [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+  val solutions = List(
+    List(-5.0,-5.0), List(-4.0, -4.0), List(-3.0, -3.0)
+  )
+  val conf = PSOConfiguration(
+    problemType = ProblemType.Minimization,
+    objectiveFunction = obj,
+    initialSolutions = solutions,
+    initialVelocity = Some(List(5.0, 5.0)),
+    topology = ParticleTopology.FullyConnected
+  )
```
Comment (kiambogo) [inline src/test/scala/SwarmMonitorSpec.scala]: Any reason to have the close parenthesis on it's own line? Is that a recommended syntax?

Comment (wittjosiah) [inline src/test/scala/SwarmMonitorSpec.scala]: you're the scala guy, you tell me :P personal preference, I find it easier to read I guess

File: src/main/scala/SwarmMonitor.scala (hunk 2 above)
Comment (wittjosiah) [inline src/main/scala/SwarmMonitor.scala]: SwarmMaster works for me, I will update

File: src/test/scala/SwarmMonitorSpec.scala (hunk 3 above)
Comment (kiambogo) [inline src/test/scala/SwarmMonitorSpec.scala]: http://docs.scala-lang.org/style/declarations.html

Put it at the end of the last line

Comment (wittjosiah) [inline src/test/scala/SwarmMonitorSpec.scala]: is that just for class declarations? what about line 45, is that fine?

Comment (kiambogo) [inline src/test/scala/SwarmMonitorSpec.scala]: http://docs.scala-lang.org/style/indentation.html (bottom)

yep

File: src/test/scala/SwarmMonitorSpec.scala [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+          )
+        )
+        expectMsg(SwarmReport(expectedState))
+      }
+    }
+  }
+}
+
+object SwarmMonitorSpec {
+
```
Comment (kiambogo) [inline src/test/scala/SwarmMonitorSpec.scala]: Remove blank lines

File: src/test/scala/SwarmMonitorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  val config = """
```
Comment (kiambogo) [inline src/test/scala/SwarmMonitorSpec.scala]: This should be moved to conf/application.conf

File: build.sbt [hunk 6]
```diff
@@ -13,7 +13,8 @@ resolvers += "Akka Snapshot Repository" at "http://repo.akka.io/snapshots/"
 libraryDependencies ++= Seq(
   "com.typesafe.akka" %% "akka-actor" % "2.4-M3",
   "com.typesafe.akka" %% "akka-testkit" % "2.4-M3",
-  "org.scalatest" % "scalatest_2.11" % "2.2.4" % "test"
+  "org.scalatest" % "scalatest_2.11" % "2.2.4" % "test",
+  "org.skife.com.typesafe.config" % "typesafe-config" % "0.3.0"
```
Comment (kiambogo) [inline build.sbt]: Did I send this to you? 
I'd feel better using the official
libraryDependencies += "com.typesafe" % "config" % "1.3.0"

File: conf/application.conf [hunk 7]
```diff
@@ -0,0 +1,4 @@
+akka {
+  loglevel = "WARNING"
+}
+
```
Comment (kiambogo) [inline conf/application.conf]: blank line

File: src/main/scala/PSO/PSOExecutor.scala [hunk 8]
```diff
@@ -30,14 +30,19 @@ class PSOExecutor(pso: PSOSystem) extends Executor {
...
+
     // termination criteria is # of iterations
     for(i <- 1 to 1000) {
       pso.particles.foreach(p => p ! ComputeIteration)
     }
+
     // wait for actors to process all messages and print results
-    Thread.sleep(1000)
-    pso.particles.foreach(p => p ! ReportRequest)
+    // Thread.sleep(1000)
```
Comment (kiambogo) [inline src/main/scala/PSO/PSOExecutor.scala]: Need this comment in? Will we use it here in the future?

**Pull request merged**